### ISSUE TITLE
Proposal: Add true/false for common Earthfile boolean flags

### ISSUE THREAD

**Issue by jclangst** [earthly/earthly#1109]
# Problem

Many of the boolean flags (e.g., `--no-cache`, `--interactive`, etc.) do not take flags. This makes it very tedious to conditionally activate these pieces of functionality. See the example below.

# Proposal

Add the ability to explicitly control flags via `--some-flag[=<true|false>]`. If the boolean value is omitted, default to `true` to preserve backwards compatibility and provide a shortcut. 

The following flags should receive this ability:

- `--push`
- `--no-cache`
- `--privileged`
- `--ssh`
- `--interactive`
- `--keep-ts`
- `--keep-own`
- `--allow-privileged`

# Examples

**Current State**
```Dockerfile
WITH_INFRA_CREDS:
  COMMAND
  ARG COMMAND
  ARG NO_CACHE=false
  ARG INTERACTIVE=false
  IF [ "$NO_CACHE" = "true" ]
    RUN --no-cache \
      --mount=type=secret,target=/tmp/.kube/config,id=+secrets/kube_config \
      --mount=type=secret,target=/tmp/.azure/accessTokens.json,id=+secrets/azure_tokens \
      --mount=type=secret,target=/tmp/.azure/azureProfile.json,id=+secrets/azure_profile \
      --mount=type=tmpfs,target=/root/.kube \
      --mount=type=tmpfs,target=/root/.azure \
      --ssh \
      -- \
      cp -R /tmp/.azure/* /root/.azure/ && \
      cp -R /tmp/.kube/* /root/.kube/ && \
      kubectl cluster-info && \
      bash -c "$COMMAND"
  ELSE IF [ "$INTERACTIVE" = "true" ]
    RUN --no-cache --interactive \
      --mount=type=secret,target=/tmp/.kube/config,id=+secrets/kube_config \
      --mount=type=secret,target=/tmp/.azure/accessTokens.json,id=+secrets/azure_tokens \
      --mount=type=secret,target=/tmp/.azure/azureProfile.json,id=+secrets/azure_profile \
      --mount=type=tmpfs,target=/root/.kube \
      --mount=type=tmpfs,target=/root/.azure \
      --ssh \
      -- \
      cp -R /tmp/.azure/* /root/.azure/ && \
      cp -R /tmp/.kube/* /root/.kube/ && \
      kubectl cluster-info && \
      bash -c "$COMMAND"
  ELSE
    RUN \
      --mount=type=secret,target=/tmp/.kube/config,id=+secrets/kube_config \
      --mount=type=secret,target=/tmp/.azure/accessTokens.json,id=+secrets/azure_tokens \
      --mount=type=secret,target=/tmp/.azure/azureProfile.json,id=+secrets/azure_profile \
      --mount=type=tmpfs,target=/root/.kube \
      --mount=type=tmpfs,target=/root/.azure \
      --ssh \
      -- \
      cp -R /tmp/.azure/* /root/.azure/ && \
      cp -R /tmp/.kube/* /root/.kube/ && \
      kubectl cluster-info && \
      bash -c "$COMMAND"
END
```

**Future State**

```Dockerfile
WITH_INFRA_CREDS:
  COMMAND
  ARG COMMAND
  ARG NO_CACHE=false
  ARG INTERACTIVE=false
  RUN --no-cache="$NO_CACHE" --interactive="$INTERACTIVE"  \
    --mount=type=secret,target=/tmp/.kube/config,id=+secrets/kube_config \
    --mount=type=secret,target=/tmp/.azure/accessTokens.json,id=+secrets/azure_tokens \
    --mount=type=secret,target=/tmp/.azure/azureProfile.json,id=+secrets/azure_profile \
    --mount=type=tmpfs,target=/root/.kube \
    --mount=type=tmpfs,target=/root/.azure \
    --ssh \
    -- \
    cp -R /tmp/.azure/* /root/.azure/ && \
    cp -R /tmp/.kube/* /root/.kube/ && \
    kubectl cluster-info && \
    bash -c "$COMMAND"
```

# Potential Alternatives

- Some form of meta-programming that allows you to dynamically construct each command

- Extract the arguments for each command into their own command `OPTION <flag>` which automatically adds the arguments to all subsequent commands; this would allow you to utilize the existing conditional mechanisms; you could clear options with `OPTION --clear`.

**Comment by alexcb** [comment]
Your before and after example is a great example of how this makes writing `COMMAND`s much easier. I'm in favour of this change.

**Comment by vladaionescu** [comment]
Good proposal!

For this part

> Some form of meta-programming that allows you to dynamically construct each command

We've been thinking something like the equivalent of the bash `eval` command. We haven't yet written a proposal for it though.

**Comment by alexcb** [comment]
@jclangst we just released [0.5.20](https://github.com/earthly/earthly/releases/tag/v0.5.20) with this feature; we would love to hear if this makes things easier for you.

**Comment by jclangst** [comment]
Awesome! I will close this and check it out. Thanks for the quick turnaround!
